src/views/search/components/search-footer.tsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from "preact/hooks";
+import style from "./search-footer.module.scss";
+
+interface Props {
+	duration: number;
+}
+
+function useDarkMode() {
+	const [isDarkTheme, setIsDarkTheme] = useState(
+		() => document.documentElement.classList.contains("dark")
+	);
+
+	const handleMutation = () => {
+		setIsDarkTheme(
+			document.documentElement.classList.contains("dark")
+		);
+	};
+
+	useEffect(() => {
+		const observer = new MutationObserver(handleMutation);
+		observer.observe(document.documentElement, { attributes: true });
+		return () => observer.disconnect();
+	}, []);
```

Comment:
Move `handleMutation` inside of `useEffect` to avoid issues with linting or stale data